packages/filecoin-address/src/index.ts

```diff
@@ -12,27 +13,33 @@ export * from './protocol'
 const defaultCoinType = CoinType.MAIN
 const base32 = base32Function('abcdefghijklmnopqrstuvwxyz234567')
 
-// PayloadHashLength defines the hash length taken over addresses using the
-// Actor and SECP256K1 protocols.
+// Defines the hash length taken over addresses
+// using the Actor and SECP256K1 protocols.
 const payloadHashLength = 20
 
+// The maximum length of a delegated address's sub-address.
+const maxSubaddressLen = 54
+
+// The number of bytes that are reserved for namespace
+const namespaceByteLen = new Int64(0).toBuffer().length
+
 function addressHash(ingest: Uint8Array): Uint8Array {
   return blake2b(ingest, null, payloadHashLength)
 }
 
 export class Address {
-  readonly str: Uint8Array
-  readonly _protocol: Protocol
+  readonly bytes: Uint8Array
   readonly _coinType: CoinType
 
-  constructor(str: Uint8Array, coinType: CoinType = defaultCoinType) {
-    if (!str || str.length < 1) throw new Error('Missing str in address')
-    this.str = str
-    this._protocol = this.str[0] as Protocol
-    if (!Protocol[this._protocol]) {
-      throw new Error(`Invalid protocol ${this._protocol}`)
-    }
+  constructor(bytes: Uint8Array, coinType: CoinType = defaultCoinType) {
+    if (!bytes || !bytes.length) throw new Error('Missing bytes in address')
+
+    this.bytes = bytes
     this._coinType = coinType
+
+    if (!(this.protocol() in Protocol)) {
+      throw new Error(`Invalid protocol ${this.protocol()}`)
+    }
   }
 
   network(): CoinType {
```

Comment:
Seems a bit unnecessary to expose all of these:
- `readonly _coinType` property (seems like it's meant to be private, but it's not)
- `network()` method returning `_coinType`
- `coinType()` method returning `coinType`

Reply:
I would keep it as-is since many are using this package. It was meant to align with go-address initially, which is why I think they were made functions

network is an artifact of poor naming premainnet